domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -3533,6 +3759,17 @@ public <T> Future<T> withConn(AsyncResult<SQLConnection> sqlConnection, Function
     }
   }
 
+  /**
+   * Take the connection from the {@link SQLConnection}, wrap it into a {@link Conn} and execute the function.
+   *
+   * @return the result from the function, the failure of sqlConnection or any thrown Throwable.
+   */
+  @SuppressWarnings("squid:S1181")  // suppress "Throwable and Error should not be caught"
+  // because a Future also handles Throwable, this is required for asynchronous reporting
+  public <T> Future<T> withReadConn(AsyncResult<SQLConnection> sqlConnection, Function<Conn, Future<T>> function) {
+    return withConn(sqlConnection, function);
+  }
```

Comment:
This method is not needed. `withConn` can directly be used.

Reply:
It's needed because of this: 
https://github.com/folio-org/raml-module-builder/pull/1047#discussion_r866189399

Reply:
In line 2540 withConn should be called because a connection is passed as a parameter that cannot be changed from writer to reader.

This method `withReadConn(AsyncResult<SQLConnection> sqlConnection, Function<Conn, Future<T>> function)` is not needed and should be deleted.